controlplane/controlplane.cpp

```diff
@@ -941,6 +953,14 @@ eResult cControlPlane::loadConfig(const std::string& rootFilePath,
 		}
 
 		YANET_LOG_INFO("dataplane has been updated (stage 7)\n");
+		for (auto& module : modules)
+		{
+			if (rib_t* rib = dynamic_cast<rib_t*>(module))
+			{
+				rib->bird_import_get();
+				rib->moduleStart();
+			}
+		}
 	}
```

Comment:
Is every module is also a rib module? I don't quite understand.
Also, `dynamic_cast` will inevitably lead to some errors, if, for example, we decide to change class hierarchy.
Can we use other things here? Like maybe we can utilize one or two virtual functions?

Reply:
At this stage, I need to call one method that is only in rib_t (bird_import_get()), using a virtual function for all classes is somehow strange. I'll think about it, or I'm ready to discuss what's best.

Reply:
`modules` is a vector of `cModule` pointers. So what you can do is add a new field to the class that will point to the required module, and here just call it's methods.
Something like an iterator to the `modules` will suffice (but be aware of invalidation). Or just a pointer to `rib_t`.
Either way it's better than what we have how.

Reply:
`modules` is a vector of `cModule` pointers. So what you can do is add a new field to the class that will point to the required module, and here just call it's methods.
Something like an iterator to the `modules` will suffice (but be aware of invalidation). Or just a pointer to `rib_t`.
Either way it's better than what we have how.